src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -4780,7 +4780,7 @@ char *wm_vuldet_cpe_str(cpe *cpe_s) {
     os_calloc(OS_SIZE_256 + 1, sizeof(char), generated_cpe);
 
     snprintf(generated_cpe, OS_SIZE_256, "%c:%s:%s:%s:%s:%s:%s:%s:%s:%s:%s",
-        *cpe_s->part,
+        cpe_s->part ? *cpe_s->part : ' ',
```

Comment:
Can you add an empty string instead of space when it is `NULL`? Just for consistency with the other fields. -> `cpe_s->part ? *cpe_s->part : ""`

Reply:
it can cause a warning because a character is expected instead of a string.